### ISSUE TITLE
SC2140 incorrectly triggered in `=~` regexp with quoted fixed strings

### ISSUE THREAD

**Issue by nhed** [koalaman/shellcheck#1565]
#### For bugs
- Rule Id: SC2140
- My shellcheck version (`shellcheck --version` or "online"):  0.6.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2140)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```bash
#!/usr/bin/env bash

declare REPODIR="/tmp/zz"

declare -a DIRELMTS=("${REPODIR}"/* "${REPODIR}"/.*)

for EL in "${DIRELMTS[@]}"; do
   if ! [[ "${EL}" =~ "${REPODIR}/"("."|".."|"somedir"|".git")$ ]]; then
       echo rm -rf "${EL}"
   fi
done

```

#### Here's what shellcheck currently says:

```log
$ shellcheck /tmp/zz/somedir/test2.sh 

In /tmp/zz/somedir/test2.sh line 8:
   if ! [[ "${EL}" =~ "${REPODIR}/"("."|".."|"somedir"|".git")$ ]]; then
                                       ^-- SC2140: Word is of the form "A"B"C" (B indicated). Did you mean "ABC" or "A\"B\"C"?
                                            ^-- SC2140: Word is of the form "A"B"C" (B indicated). Did you mean "ABC" or "A\"B\"C"?
                                                      ^-- SC2140: Word is of the form "A"B"C" (B indicated). Did you mean "ABC" or "A\"B\"C"?

```

#### Here's what I wanted or expected to see:

Expected to see no complaints ... looks like legit code to me
matching a regexp where I quote all the fixed strings (especially important with `.` and `..`

sure I can write it with `case` but that's more last millennium 

If nothing else, message should be clearer

**Comment by koalaman** [comment]
You're right of course, this has now been fixed. ShellCheck parses the `()|`s in `=~` regex as simple literals even though they're not, and that caused this warning to overtrigger. Thanks!
